dash/include/dash/iterator/GlobIter.h

```diff
@@ -321,6 +321,10 @@ class GlobIter : public std::iterator<
       DASH_LOG_TRACE_VAR("GlobIter.dart_gptr", idx);
       DASH_LOG_TRACE_VAR("GlobIter.dart_gptr", offset);
     }
+
+    if (offset == 1) {
```

Comment:
What is this for?

Reply:
Good question. In general, the implementation of `GlobIter` is really awkward and we cannot simply fix it because many other place in our code rely on this. Will invest more research to figure it out. The whole thing looks to me like broken state machine.

Reply:
Will push more changes later today / tomorrow.

Reply:
Should look better now. I removed duplicate code.